roles/libvirt_manager/tasks/attach_interface.yml

```diff
@@ -73,10 +73,12 @@
         (_net_name not in _net_bridge_map.keys()) |
         ternary(_net_name, _net_bridge_map[_net_name])
       }}
+    _type: "{{ cifmw_libvirt_manager_network_interface_types[network.name] | default('bridge') }}"
     _attached_bridges: >-
       {{
         _extracted_xml.matches | default([]) |
-        selectattr('source.bridge', 'equalto', _local_bridge_name) | default([])
+        selectattr('source.' + _type, 'defined') | default([]) |
+        selectattr('source.' + _type, 'equalto', _local_bridge_name) | default([])
```

Comment:
Those `| default([])` are not needed. It's safe to iterate the list/iterator `selectattr` returns in case it's empty.